ethers-solc/src/compile.rs

```diff
@@ -225,6 +225,30 @@ impl Solc {
         Ok(())
     }
 
+    /// Verify that the checksum for this version of solc is correct. We check against the SHA256
+    /// checksum from the build information published by binaries.soliditylang
+    #[cfg(all(feature = "svm", feature = "async"))]
+    pub fn verify_checksum(&self) -> Result<()> {
+        let version = self.version_short()?;
+        let mut version_path = svm::version_path(version.to_string().as_str());
+        version_path.push(format!("solc-{}", version.to_string().as_str()));
+        let content = std::fs::read(version_path)?;
+
+        use sha2::Digest;
+        let mut hasher = sha2::Sha256::new();
+        hasher.update(&content);
+        let checksum_calc = &hasher.finalize()[..];
+
+        let all_releases = tokio::runtime::Runtime::new().unwrap().block_on(svm::all_releases(svm::platform()))?;
```

Comment:
Should probably use [`RELEASES`](https://github.com/gakonst/ethers-rs/blob/8e09715125f4aad77f43172fde02a83d2d91af2f/ethers-solc/src/compile.rs#L49) instead?

Reply:
I was trying to get rid of this duplication. Actually `RELEASES` fetches only a sorted `Vec<Version>` (basically the `svm::all_versions`).

What I used here is `svm::all_releases` which also provides the build info (checksum info).

Reply:
In any case, it'd be nice if we cached the response in a static or in `self` so that we don't need to query it each time?

Reply:
yeah, I think a global Lazy would be appropriate here, within the lazy init we can use this pattern here to get the runtime https://github.com/gakonst/foundry/blob/432e91196138f63289be40424599869f4d60f208/evm-adapters/src/blocking_provider.rs#L17-L20

Reply:
Yup, gonna refactor that part. Sounds good!